contracts/marketpools.js

```diff
@@ -199,21 +206,42 @@ actions.addLiquidity = async (payload) => {
   if (api.assert(pool, 'no existing pool for tokenPair')) {
     if (api.BigNumber(pool.baseQuantity).eq(0) && api.BigNumber(pool.quoteQuantity).eq(0)
       && await validateOracle(pool, api.BigNumber(quoteQuantity).dividedBy(baseQuantity)) === false) return;
-    if (api.BigNumber(pool.baseQuantity).gt(0) && api.BigNumber(pool.quoteQuantity).gt(0)
-      && !validateLiquiditySwap(pool, baseQuantity, quoteQuantity)) return;
+
+    let amountAdjusted;
+    const baseMin = api.BigNumber(baseQuantity).times(api.BigNumber('1').minus(addSlippage));
+    const quoteMin = api.BigNumber(quoteQuantity).times(api.BigNumber('1').minus(addSlippage));
+    if (api.BigNumber(pool.baseQuantity).gt(0) && api.BigNumber(pool.quoteQuantity).gt(0)) {
+      const quoteOptimal = getQuote(baseQuantity, pool.baseQuantity, pool.quoteQuantity).toFixed(quoteToken.precision, api.BigNumber.ROUND_HALF_UP);
+      if (api.BigNumber(quoteOptimal).lte(quoteQuantity)) {
+        if (!api.assert(api.BigNumber(quoteOptimal).gte(quoteMin), 'exceeded max slippage for adding liquidity')) return;
+        amountAdjusted = [baseQuantity, quoteOptimal];
+      } else {
+        const baseOptimal = getQuote(quoteQuantity, pool.quoteQuantity, pool.baseQuantity).toFixed(baseToken.precision, api.BigNumber.ROUND_HALF_UP);
+        if (api.BigNumber(baseOptimal).lte(baseQuantity)) {
+          if (!api.assert(api.BigNumber(baseOptimal).gte(baseMin), 'exceeded max slippage for adding liquidity')) return;
+          amountAdjusted = [baseOptimal, quoteQuantity];
+        }
+      }
+      const pPrecision = Math.min(baseToken.precision, quoteToken.precision);
```

Comment:
Want to re-raise this, but in terms of precision, should it not be by the token unit of this price (either quote or base, not sure which)